### DISCUSSION TITLE
Upgrade to 2024.08 failed

### DISCUSSION THREAD

**Discussion by crefeldtc** [uyuni-project/uyuni#9301] [General]
I tried to upgrade from 2024.03 to 2024.08, following the advise at [https://www.uyuni-project.org/uyuni-docs/en/uyuni/installation-and-upgrade/server-minor-upgrade-uyuni.html](https://www.uyuni-project.org/uyuni-docs/en/uyuni/installation-and-upgrade/server-minor-upgrade-uyuni.html) and there was no error reported by the scripts.

Afterwards I replaced the entries for the SSL-certificates as usual and rebooted the system.

Now calling [https://uyuni.lkvbay.net/](https://uyuni.lkvbay.net/) delivers immediately an 

> HTTP Status 404: 
> Beschreibung The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.
> Apache Tomcat/9.0.91
> 

After a spacewalk-service stop and start system returns FAILED and checking spacewalk-service status shows that every service is running fine with one exception:

> × salt-secrets-config.service - Configures secrets between salt-master and other services
>      Loaded: loaded (/usr/lib/systemd/system/salt-secrets-config.service; static)
>     Drop-In: /usr/lib/systemd/system/salt-secrets-config.service.d
>              └─override.conf
>      Active: failed (Result: exit-code) since Thu 2024-09-26 14:56:15 CEST; 22min ago
>     Process: 12847 ExecStart=/usr/bin/salt-secrets-config.py (code=exited, status=1/FAILURE)
>    Main PID: 12847 (code=exited, status=1/FAILURE)
> 

Any idea how to fix it?

Thanks in advance!

Regards, 
 Tobias Crefeld.

**Comment by spehyr** [comment]
The same here but the difference is that i did  have 2024.05  and what i did was just zypper ref - zypper up 
spacewalk-service status
● uyuni-update-config.service - Uyuni update config
     Loaded: loaded (/usr/lib/systemd/system/uyuni-update-config.service; static)
     Active: active (exited) since Thu 2024-10-03 10:38:04 CEST; 6min ago
    Process: 6834 ExecStart=/usr/sbin/uyuni-update-config (code=exited, status=0/SUCCESS)
   Main PID: 6834 (code=exited, status=0/SUCCESS)

● uyuni-check-database.service - Uyuni check database
     Loaded: loaded (/usr/lib/systemd/system/uyuni-check-database.service; static)
     Active: active (exited) since Thu 2024-10-03 10:38:04 CEST; 6min ago
    Process: 6870 ExecStart=/usr/sbin/spacewalk-startup-helper check-database (code=exited, status=0/SUCCESS)
   Main PID: 6870 (code=exited, status=0/SUCCESS)

● tomcat.service - Apache Tomcat Web Application Container
     Loaded: loaded (/usr/lib/systemd/system/tomcat.service; enabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/tomcat.service.d
             └─override.conf
     Active: active (running) since Thu 2024-10-03 10:38:04 CEST; 6min ago
   Main PID: 6989 (java)
      Tasks: 78 (limit: 576)
     CGroup: /system.slice/tomcat.service
             └─ 6989 /usr/lib64/jvm/jre/bin/java -ea -Xms256m -Xmx1G -Djava.awt.headless=true -Dorg.xml.sax.driver=com.redhat.rhn.frontend.xmlrpc.util.RhnSAX…

● spacewalk-wait-for-tomcat.service - Spacewalk wait for tomcat
     Loaded: loaded (/usr/lib/systemd/system/spacewalk-wait-for-tomcat.service; static)
     Active: active (exited) since Thu 2024-10-03 10:38:18 CEST; 6min ago
    Process: 6990 ExecStart=/usr/sbin/spacewalk-startup-helper wait-for-tomcat (code=exited, status=0/SUCCESS)
   Main PID: 6990 (code=exited, status=0/SUCCESS)

● salt-master.service - The Salt Master Server
     Loaded: loaded (/usr/lib/systemd/system/salt-master.service; enabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/salt-master.service.d
             └─override.conf
     Active: active (running) since Thu 2024-10-03 10:38:04 CEST; 6min ago
       Docs: man:salt-master(1)
             file:///usr/share/doc/salt/html/contents.html
             https://docs.saltproject.io/en/latest/contents.html
   Main PID: 6987 (salt-master)
      Tasks: 55
     CGroup: /system.slice/salt-master.service
             ├─ 6987 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7173 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7193 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7225 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7226 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7227 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7228 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7230 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7231 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7232 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7233 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7234 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7235 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7236 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7246 /usr/bin/python3 /usr/bin/salt-master
             ├─ 7247 /usr/bin/python3 /usr/bin/salt-master
             └─ 7249 /usr/bin/python3 /usr/bin/salt-master

● salt-api.service - The Salt API
     Loaded: loaded (/usr/lib/systemd/system/salt-api.service; enabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/salt-api.service.d
             └─override.conf
     Active: active (running) since Thu 2024-10-03 10:38:04 CEST; 6min ago
       Docs: man:salt-api(1)
             file:///usr/share/doc/salt/html/contents.html
             https://docs.saltproject.io/en/latest/contents.html
   Main PID: 6982 (salt-api)
      Tasks: 104 (limit: 4915)
     CGroup: /system.slice/salt-api.service
             ├─ 6982 /usr/bin/python3 /usr/bin/salt-api
             └─ 7086 /usr/bin/python3 /usr/bin/salt-api

● spacewalk-wait-for-salt.service - Make sure that salt is started before httpd
     Loaded: loaded (/usr/lib/systemd/system/spacewalk-wait-for-salt.service; static)
     Active: active (exited) since Thu 2024-10-03 10:38:04 CEST; 6min ago
    Process: 6988 ExecStart=/bin/true (code=exited, status=0/SUCCESS)
   Main PID: 6988 (code=exited, status=0/SUCCESS)

● apache2.service - The Apache Webserver
     Loaded: loaded (/usr/lib/systemd/system/apache2.service; enabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/apache2.service.d
             └─override.conf
     Active: active (running) since Thu 2024-10-03 10:38:18 CEST; 6min ago
   Main PID: 8580 (httpd-prefork)
     Status: "Processing requests..."
      Tasks: 8
     CGroup: /system.slice/apache2.service
             ├─ 8580 /usr/sbin/httpd-prefork -DSYSCONFIG -DSSL -DISSUSE -C "PidFile /run/httpd.pid" -C "Include /etc/apache2/sysconfig.d//loadmodule.conf" -C…
             ├─ 8593 /usr/sbin/httpd-prefork -DSYSCONFIG -DSSL -DISSUSE -C "PidFile /run/httpd.pid" -C "Include /etc/apache2/sysconfig.d//loadmodule.conf" -C…
             ├─ 8594 /usr/sbin/httpd-prefork -DSYSCONFIG -DSSL -DISSUSE -C "PidFile /run/httpd.pid" -C "Include /etc/apache2/sysconfig.d//loadmodule.conf" -C…
             ├─ 8595 /usr/sbin/httpd-prefork -DSYSCONFIG -DSSL -DISSUSE -C "PidFile /run/httpd.pid" -C "Include /etc/apache2/sysconfig.d//loadmodule.conf" -C…
             ├─ 8596 /usr/sbin/httpd-prefork -DSYSCONFIG -DSSL -DISSUSE -C "PidFile /run/httpd.pid" -C "Include /etc/apache2/sysconfig.d//loadmodule.conf" -C…
             ├─ 8598 /usr/sbin/httpd-prefork -DSYSCONFIG -DSSL -DISSUSE -C "PidFile /run/httpd.pid" -C "Include /etc/apache2/sysconfig.d//loadmodule.conf" -C…
             ├─ 8791 /usr/sbin/httpd-prefork -DSYSCONFIG -DSSL -DISSUSE -C "PidFile /run/httpd.pid" -C "Include /etc/apache2/sysconfig.d//loadmodule.conf" -C…
             └─ 8927 /usr/sbin/httpd-prefork -DSYSCONFIG -DSSL -DISSUSE -C "PidFile /run/httpd.pid" -C "Include /etc/apache2/sysconfig.d//loadmodule.conf" -C…

● rhn-search.service - Spacewalk search engine
     Loaded: loaded (/usr/lib/systemd/system/rhn-search.service; disabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/rhn-search.service.d
             └─override.conf
     Active: active (running) since Thu 2024-10-03 10:38:18 CEST; 6min ago
   Main PID: 8586 (rhn-search)
      Tasks: 83 (limit: 4915)
     CGroup: /system.slice/rhn-search.service
             ├─ 8586 /bin/sh /usr/sbin/rhn-search
             └─ 8592 /usr/bin/java -Djava.library.path=/usr/lib:/usr/lib64:/usr/lib/gcj/postgresql-jdbc:/usr/lib64/gcj/postgresql-jdbc -classpath "/usr/share…

● cobblerd.service - Cobbler Helper Daemon
     Loaded: loaded (/usr/lib/systemd/system/cobblerd.service; enabled; vendor preset: disabled)
     Active: active (running) since Thu 2024-10-03 10:38:05 CEST; 6min ago
   Main PID: 6980 (cobblerd)
      Tasks: 1 (limit: 4915)
     CGroup: /system.slice/cobblerd.service
             └─ 6980 /usr/bin/python3.6 -s /usr/bin/cobblerd -F

● taskomatic.service - Taskomatic
     Loaded: loaded (/usr/lib/systemd/system/taskomatic.service; enabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/taskomatic.service.d
             └─override.conf
     Active: active (running) since Thu 2024-10-03 10:38:18 CEST; 6min ago
   Main PID: 8587 (taskomatic)
      Tasks: 151 (limit: 4915)
     CGroup: /system.slice/taskomatic.service
             ├─ 8587 /bin/sh /usr/sbin/taskomatic
             └─ 8606 /usr/bin/java -Djava.library.path=/usr/lib:/usr/lib64 -classpath "/usr/share/rhn/classes:/usr/share/rhn/lib/spacewalk-asm.jar:/usr/share…

● spacewalk-wait-for-taskomatic.service - Spacewalk wait for taskomatic
     Loaded: loaded (/usr/lib/systemd/system/spacewalk-wait-for-taskomatic.service; static)
     Active: active (exited) since Thu 2024-10-03 10:38:21 CEST; 6min ago
    Process: 8599 ExecStart=/usr/sbin/spacewalk-startup-helper wait-for-taskomatic (code=exited, status=0/SUCCESS)
   Main PID: 8599 (code=exited, status=0/SUCCESS)

× salt-secrets-config.service - Configures secrets between salt-master and other services
     Loaded: loaded (/usr/lib/systemd/system/salt-secrets-config.service; static)
    Drop-In: /usr/lib/systemd/system/salt-secrets-config.service.d
             └─override.conf
     Active: failed (Result: exit-code) since Thu 2024-10-03 10:38:04 CEST; 6min ago
    Process: 6832 ExecStart=/usr/bin/salt-secrets-config.py (code=exited, status=1/FAILURE)
   Main PID: 6832 (code=exited, status=1/FAILURE)

● mgr-websockify.service - TCP to WebSocket proxy
     Loaded: loaded (/usr/lib/systemd/system/mgr-websockify.service; static)
     Active: active (running) since Thu 2024-10-03 10:38:04 CEST; 6min ago
    Process: 6981 ExecStartPre=/usr/bin/sh -c grep secret_key /etc/rhn/rhn.conf | tr -d ' ' | cut -f2 -d '=' | perl -ne 's/([0-9a-f]{2})/print chr hex $1/gie' > /etc/rhn/websockify.key (code=exited, status=0/SUCCESS)
   Main PID: 6991 (websockify)
      Tasks: 8 (limit: 4915)
     CGroup: /system.slice/mgr-websockify.service
             └─ 6991 /usr/bin/python3 /usr/bin/websockify --token-plugin JWTTokenApi --token-source /etc/rhn/websockify.key localhost:8050

● cobbler-refresh-mkloaders.service - Refresh Cobbler bootloaders
     Loaded: loaded (/usr/lib/systemd/system/cobbler-refresh-mkloaders.service; static)
    Drop-In: /usr/lib/systemd/system/cobbler-refresh-mkloaders.service.d
             └─override.conf
     Active: active (exited) since Thu 2024-10-03 10:38:19 CEST; 6min ago
    Process: 8591 ExecStart=/usr/bin/cobbler mkloaders (code=exited, status=0/SUCCESS)
   Main PID: 8591 (code=exited, status=0/SUCCESS)
○ spacewalk.target - Spacewalk
     Loaded: loaded (/usr/lib/systemd/system/spacewalk.target; enabled; vendor preset: disabled)
     Active: inactive (dead)
